### ISSUE TITLE
hostnames do not match IP after RPi upgrade

### ISSUE THREAD

**Issue by Marooned-MB** [pi-hole/FTL#924]
### Versions

- Pi-hole:  v5.1.2 (Latest: v5.1.2)
- AdminLTE:  v5.1.1 (Latest: v5.1.1)
- FTL:  v5.2 (Latest: v5.2)

### Platform

- OS and version:  Linux raspberrypi 5.4.72+ pi-hole/pi-hole#1356 Thu Oct 22 13:56:00 BST 2020 armv6l GNU/Linux, 10 (buster)

- Platform:  Raspberry Pi Model B Rev 2 (512MB)

### Expected behavior

I had pi-hole running on old Raspbian and all was good. Few days ago I decided to wipe out my SD card and install newest soft for my RPi. I've installed latest OS and other programs. After setting up pi-hole I've noticed that it assigned wrong hostnames to some IPs (see attachment - green ones are correct, red one are not).
What is probably worth noting, I copied `/etc/.pihole` and `/etc/pihole` from old system to restore the backup and all my settings/white/blacklists.

### Actual behavior / bug

pi-hole assigns wrong hostnames to some IP resulting in some strange results like the vacuum cleaner calling Facebook etc

Also, color of the background in `/admin/network.php` suggest that those devices were not seen in >24h which is not true as `/admin/queries.php` shows proper and updated list (just wrong hostnames).

### Steps to reproduce

Just checking different pages in pi.hole/admin

## Debug Token

- URL: https://tricorder.pi-hole.net/6i4xc8ma1n

### Screenshots

![pi-hole ip](https://user-images.githubusercontent.com/716018/99062939-06919300-25a4-11eb-94f3-51fcb28c81cb.jpg)

**Comment by DL6ER** [comment]
What is the output of
```
ip neigh show
```
?

**Comment by Marooned-MB** [comment]
`ip neigh show` shows a list of IPs and MAC address and they match the list from the router, so the proper one.

**Comment by DL6ER** [comment]
Then something odd may have happened in between. Please try flushing your Pi-hole's address assignments using the command below. You will not loose any other details like the first seen timestamps, etc.

```
sudo sqlite3 /etc/pihole/pihole-FTL.db "DELETE FROM network_addresses;"
```

Then wait some minutes and check again what the Pi-hole table shows to you.

**Comment by Marooned-MB** [comment]
After running that query IP column on `/admin/network.php` went blank. Also, `Dashboard` was wiped out. I waited ~20 minutes and no IP address has been updated in the Network table. So I went to `Settings` and hit `Flush network table`. Now the whole table on `network.php` has been cleared (as expected) but no new entry shows up. In the meantime `Query log` shows real time updates on allowed and blocked domains, so pi-hole is running correctly. `Dashboard` is slowly filling in.

**Comment by DL6ER** [comment]
So it seems `pihole-FTL` isn't able to read any data from `ip neigh show` and, hence, also the self-healing didn't set in. Have you tried restarting the machine you're running Pi-hole on? Before we start taking apart your system piece-by-piece, it'd be great to outrule all other possible causes.

**Comment by Marooned-MB** [comment]
Yes, I did many restarts, the last one was after running above query, just to make sure pi-hole is picking up latest data.

Is the ownership of the files correct?
```
pi@raspberrypi:~ $ ls -l /etc/pihole
total 634160
-rw-r--r-- 1 pihole pihole         0 Nov 15 18:31 dhcp.leases
-rw-r--r-- 1 root   root         596 May 23 16:23 dns-servers.conf
-rw-r--r-- 1 root   root          18 Nov 15 18:31 GitHubVersions
-rw-rw-r-- 1 pihole pihole   5423104 Nov 15 03:58 gravity.db
-rw-r--r-- 1 root   root        1208 May 23 16:23 install.log
-rw-r--r-- 1 root   root     1165238 Nov 15 03:58 list.0.raw.githubusercontent.com.domains
-rw-r--r-- 1 root   root      594616 Aug 16 02:33 list.1.mirror1.malwaredomains.com.domains
-rw-r--r-- 1 root   root         521 May 23 16:24 list.2.s3.amazonaws.com.domains
-rw-r--r-- 1 root   root       43529 May 23 16:24 list.3.s3.amazonaws.com.domains
-rw-r--r-- 1 root   root       52866 Nov 15 03:58 list.4.raw.githubusercontent.com.domains
-rw-r--r-- 1 root   root          20 Nov 15 21:10 localbranches
-rw-r--r-- 1 root   root          48 Nov 15 03:58 local.list
-rw-r--r-- 1 root   root          46 Nov 15 21:10 localversions
-rw-r--r-- 1 root   root         234 May 23 16:23 logrotate
-rw-r--r-- 1 root   root     2457600 May 23 16:23 macvendor.db
drwxr-xr-x 2 root   root        4096 Nov  6 18:16 migration_backup
-rw-r--r-- 1 root   root          15 May 23 16:23 pihole-FTL.conf
-rw-r--r-- 1 pihole pihole 319791104 Nov 15 18:43 pihole-FTL.db
-rw-r--r-- 1 root   root         408 Nov 13 17:32 setupVars.conf
```

**Comment by DL6ER** [comment]
Yeah, that seems fine at first glance. For your issue, only the permissions of the `pihole-FTL.db` file should be important and that seems to be correct.

Does 
```
sqlite3 /etc/pihole/pihole-FTL.db "SELECT * FROM network_addresses;"
```
produce any output?

Also, are there any errors being reported in `/var/log/pihole-FTL.log` ?

**Comment by Marooned-MB** [comment]
```
pi@raspberrypi:~ $ sqlite3 /etc/pihole/pihole-FTL.db "SELECT * FROM network_addresses;"
pi@raspberrypi:~ $ tail /var/log/pihole-FTL.log
[2020-11-15 18:31:11.814 482M] New upstream server: 1.0.0.1 (0/1024)
[2020-11-15 18:31:37.996 482M] New upstream server: 1.1.1.1 (1/1024)
[2020-11-15 18:41:03.193 482M] Resizing "/FTL-strings" from 4096 to 8192
[2020-11-15 18:45:02.501 482M] Resizing "/FTL-dns-cache" from 4096 to 8192
[2020-11-15 18:58:02.859 482M] Resizing "/FTL-strings" from 8192 to 12288
[2020-11-15 19:24:40.345 482M] Resizing "/FTL-dns-cache" from 8192 to 12288
[2020-11-15 19:29:10.868 482M] Resizing "/FTL-strings" from 12288 to 16384
[2020-11-15 20:24:21.748 482M] Resizing "/FTL-dns-cache" from 12288 to 16384
[2020-11-15 20:51:44.755 482M] Resizing "/FTL-queries" from 229376 to 458752
[2020-11-15 21:32:22.009 482M] Resizing "/FTL-strings" from 16384 to 20480
```

**Comment by Marooned-MB** [comment]
I decided to check what's in the tables:
```
pi@raspberrypi:~ $ for i in `sudo sqlite3 /etc/pihole/pihole-FTL.db "SELECT name FROM sqlite_master WHERE
type='table';"`; do echo -n $i\ \=\ ; sudo sqlite3 /etc/pihole/pihole-FTL.db "SELECT
COUNT(*) FROM $i;"; done
queries = Error: database disk image is malformed
sqlite_sequence = 1
ftl = 3
counters = 2
network_addresses = 0
network = 0
message = 0
```
Aha, some errors.. time to investigate:
```
pi@raspberrypi:~ $ sudo sqlite3 /etc/pihole/pihole-FTL.db
SQLite version 3.27.2 2019-02-25 16:06:06
Enter ".help" for usage hints.
sqlite> pragma integrity_check;
*** in database main ***
On tree page 77886 cell 0: invalid page number 78074
On tree page 77886 cell 352: 2nd reference to page 78072
On tree page 77499 cell 0: invalid page number 78073
Error: database disk image is malformed
```
Following [that article](https://techblog.dorogin.com/sqliteexception-database-disk-image-is-malformed-77e59d547c50?gi=5176a9bb601c) I've recreated db from a dump only to see an empty db after >1h of dumping and loading data.. I've checked dump.sql and the last line was `ROLLBACK; -- due to errors`, great...

Anyway, as a last resort I removed `pihole-FTL.db` and restarted RPi. Now the table in `network.php` starts to fill in however each rows has `unknown` in Hostname column. Also, what is surprising, two rows have tick in `Uses pi-hole` column but are in red indicating that they do not use pi-hole - a bit confusing.

![obraz](https://user-images.githubusercontent.com/716018/99200233-6bb3d700-27a4-11eb-8f75-20e8dadeb380.png)

**Comment by Marooned-MB** [comment]
Just an update after night. 10 registered devices in Network page, around ~12k queries, all still have `unknown` in `Hostname` column. Any hint?

**Comment by Marooned-MB** [comment]
I've restarted RPi and flushed network tables few times more to see if something interesting will shop up in logs.

I've checked source of [network-table.c](https://github.com/pi-hole/FTL/blob/master/src/database/network-table.c) and [resolve.c](https://github.com/pi-hole/FTL/blob/master/src/resolve.c) and I see lots of possibly useful logs done by `logg()` function. Where is that log? Maybe it will help to find the problem?

After recent network table flush RPi itself appeared on the list and it's the only device with known hostname. MAC address is empty but I assume there is a different logic for the device running pi-hole.
![obraz](https://user-images.githubusercontent.com/716018/99272761-74e68780-2828-11eb-8b91-41c563f6acae.png)

**Comment by DL6ER** [comment]
> Where is that log? Maybe it will help to find the problem?

`/var/log/pihole-FTL.log`

You'll have to add
```
DEBUG_ARP=true
DEBUG_RESOLVER=true
```
to `/etc/pihole/pihole-FTL.conf` to enable them (create the file if it does not exist). Run `pihole restartdns` after adding these lines.

- The Pi-hole itself showing up without a MAC address is correct, there is no need for an entry in the ARP cache to get to an address of the local machine. Hence, we cannot source the MAC address from anywhere (at least not without any meaningful amount of extra work)
- The host names missing from there seems to be a separate issue (it can resolve local host names, but no foreign ones). Can your Pi-hole resolve host names? Run, e.g., `dig -x 192.168.1.30` to see if anything comes out. If not, then neither your Pi-hole is responsible for DNS nor did you add a suitable upstream DNS server to ask for this.
- The tick "Uses Pi-hole" for the device which apparently never sent any query to your Pi-hole is strange. However, I guess this is a comestic bug. Let's see if this resolves itself. If so, then we probably take the wrong timestamp ("first seen" where it should have been "last query").

**Comment by Marooned-MB** [comment]
```
pi@raspberrypi:~ $ dig -x 192.168.1.30

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.30
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 36443
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;30.1.168.192.in-addr.arpa.     IN      PTR

;; Query time: 13 msec
;; SERVER: 1.1.1.1#53(1.1.1.1)
;; WHEN: Tue Nov 17 09:16:18 GMT 2020
;; MSG SIZE  rcvd: 54
```

> If not, then neither your Pi-hole is responsible for DNS nor did you add a suitable upstream DNS server to ask for this.

I have RPi IP set as DNS in my FritzBox router. Ads are blocked, pi-hole is catching all visited domains. While RPi is restarting, I can't access any domains from other devices. Looks to me that pi-hole is the DNS.

After enabling debugging this is what poped out:
```
[2020-11-17 09:26:00.526 13735/T13741] Trying to resolve 192.168.1.30
[2020-11-17 09:26:00.527 13735/T13741] Setting nameservers to:
[2020-11-17 09:26:00.527 13735/T13741]  0: 127.0.0.1:53
[2020-11-17 09:26:00.527 13735/T13741]  1: 0.0.0.0:53
[2020-11-17 09:26:00.527 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.532 13735/T13741] Setting nameservers back to default:
[2020-11-17 09:26:00.533 13735/T13741]  0: 1.1.1.1:53
[2020-11-17 09:26:00.533 13735/T13741]  1: 1.0.0.1:53
[2020-11-17 09:26:00.533 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.545 13735/T13741]  ---> "" (N/A)
[2020-11-17 09:26:00.550 13735/T13741] Trying to resolve 192.168.1.36
[2020-11-17 09:26:00.550 13735/T13741] Setting nameservers to:
[2020-11-17 09:26:00.550 13735/T13741]  0: 127.0.0.1:53
[2020-11-17 09:26:00.551 13735/T13741]  1: 0.0.0.0:53
[2020-11-17 09:26:00.551 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.556 13735/T13741] Setting nameservers back to default:
[2020-11-17 09:26:00.556 13735/T13741]  0: 1.1.1.1:53
[2020-11-17 09:26:00.556 13735/T13741]  1: 1.0.0.1:53
[2020-11-17 09:26:00.557 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.569 13735/T13741]  ---> "" (N/A)
[2020-11-17 09:26:00.574 13735/T13741] Trying to resolve 127.0.0.1
[2020-11-17 09:26:00.575 13735/T13741] Setting nameservers to:
[2020-11-17 09:26:00.575 13735/T13741]  0: 127.0.0.1:53
[2020-11-17 09:26:00.575 13735/T13741]  1: 0.0.0.0:53
[2020-11-17 09:26:00.575 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.576 13735/T13741]  ---> "localhost" (found internally)
[2020-11-17 09:26:00.576 13735/T13741] Setting nameservers back to default:
[2020-11-17 09:26:00.576 13735/T13741]  0: 1.1.1.1:53
[2020-11-17 09:26:00.576 13735/T13741]  1: 1.0.0.1:53
[2020-11-17 09:26:00.577 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.577 13735/T13741] Trying to resolve 192.168.1.245
[2020-11-17 09:26:00.577 13735/T13741] Setting nameservers to:
[2020-11-17 09:26:00.577 13735/T13741]  0: 127.0.0.1:53
[2020-11-17 09:26:00.578 13735/T13741]  1: 0.0.0.0:53
[2020-11-17 09:26:00.578 13735/T13741]  2: 0.0.0.0:0
[2020-11-17 09:26:00.584 13735/T13741]  ---> "raspberrypi" (found internally)
[2020-11-17 09:26:00.585 13735/T13741] Setting nameservers back to default:
[2020-11-17 09:26:00.585 13735/T13741]  0: 1.1.1.1:53
[2020-11-17 09:26:00.585 13735/T13741]  1: 1.0.0.1:53
[2020-11-17 09:26:00.585 13735/T13741]  2: 0.0.0.0:0
```

**Comment by Marooned-MB** [comment]
Is there anything more that could be done to fix this issue?
Thanks

**Comment by DL6ER** [comment]
> Looks to me that pi-hole is the DNS.

Yeah, sorry, I see now that I phrased this incorrectly. What I meant is that Pi-hole isn't able to get host names because nobody it could ask knows them. Hence, your call to `$ dig -x 192.168.1.30` returned an empty domain.

Try setting up Conditional Forwarding to your Fritzbox. This should allow you to get the host names of your devices from the DHCP server of your Fritzbox.

**Comment by Marooned-MB** [comment]
`Conditional Forwarding` was what I needed. Not sure why I haven't found it before while Googling and clicking around pi-hole.

There are still 6 (out of 14) `unknown` (in router they have proper hostnames), but the rest looks ok. Maybe UI should have manual edit field where I could put a hostname in such cases. SQL update on DB can't be done as DB is in read only mode and as far as I know, there is no command to disable pi-hole service?

Anyway, it's not perfect (not all devices were recognized) but not sure if we can do anything about it at this point? If not, we can close the issue as "partially fixed".

**Comment by DL6ER** [comment]
> not sure if we can do anything about it at this point

If the router knows host names for them and the router works correctly (as in: it replies to name requests), they should all show up. Try some more `dig -x 192.168.1.x` with some IP addresses of the devices which are still `unknown`. 

> as DB is in read only mode

What is the exact error message you're seeing? The database is owned by user `pihole` so it's expected that you cannot write to it as your regular user. If you still cannot write to it equipped with `sudo`-powers, then that's another thing we should look at.

We'll find a solution to this all in the end :-)

**Comment by Marooned-MB** [comment]
My `Conditional Forwarding` config in pi-hole:

Local network in CIDR notation | IP address of your DHCP server (router) | Local domain name (optional)
-- | -- | --
192.168.1.0/24 | 192.168.1.1 | fritz.box

IP which was recognized (Hostname in pi-hole: `wdmycloud.fritz.box`, Name in router: `WDMyCloud`)
```
pi@raspberrypi:~ $ dig -x 192.168.1.125

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.125
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 51335
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;125.1.168.192.in-addr.arpa.    IN      PTR

;; Query time: 11 msec
;; SERVER: 1.1.1.1#53(1.1.1.1)
;; WHEN: Mon Nov 23 15:18:49 GMT 2020
;; MSG SIZE  rcvd: 55
```

IP which was **not** recognized (Hostname in pi-hole: `unknown`, Name in router: `YI-cam`)
```
pi@raspberrypi:~ $ dig -x 192.168.1.32

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.32
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 28840
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;32.1.168.192.in-addr.arpa.     IN      PTR

;; Query time: 8 msec
;; SERVER: 1.1.1.1#53(1.1.1.1)
;; WHEN: Mon Nov 23 15:19:33 GMT 2020
;; MSG SIZE  rcvd: 54
```

_PS (not related, just side info)
Out of curiosity I picked 7 or 30 days in db_queries.php and.. after >1h I had to hard reset RPi. It took all resources, so I was not able to ssh into it, reset and I couldn't use Internet as usual (because of no DNS). I guess it would be nice (if possible) to cap CPU usage on this heavy query._

**Comment by DL6ER** [comment]
Oh, I missed

> `;; SERVER: 1.1.1.1#53(1.1.1.1)`

Could you run the same again explicitly querying your Pi-hole (`dig -x 192.168.1.x @127.0.0.1`) and also once querying your Fritzbox (`dig -x 192.168.1.x @192.168.1.1`) ?

**Comment by Marooned-MB** [comment]
```
pi@raspberrypi:~ $ dig -x 192.168.1.125 @127.0.0.1

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.125 @127.0.0.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60522
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 1, ADDITIONAL: 1

;; QUESTION SECTION:
;125.1.168.192.in-addr.arpa.    IN      PTR

;; ANSWER SECTION:
125.1.168.192.in-addr.arpa. 9   IN      PTR     WDMyCloud.fritz.box.

;; AUTHORITY SECTION:
125.1.168.192.in-addr.arpa. 9   IN      NS      fritz.box.

;; ADDITIONAL SECTION:
fritz.box.              9       IN      A       192.168.1.1

;; Query time: 8 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Mon Nov 23 19:55:53 GMT 2020
;; MSG SIZE  rcvd: 107
```
```
pi@raspberrypi:~ $ dig -x 192.168.1.32 @127.0.0.1

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.32 @127.0.0.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 53980
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0

;; QUESTION SECTION:
;32.1.168.192.in-addr.arpa.     IN      PTR

;; AUTHORITY SECTION:
1.168.192.in-addr.arpa. 9       IN      SOA     fritz.box. admin.fritz.box. 1606161362 21600 1800 43200 10

;; Query time: 4 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Mon Nov 23 19:56:02 GMT 2020
;; MSG SIZE  rcvd: 103
```
```
pi@raspberrypi:~ $ dig -x 192.168.1.125 @192.168.1.1

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.125 @192.168.1.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 19362
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 1, ADDITIONAL: 1

;; QUESTION SECTION:
;125.1.168.192.in-addr.arpa.    IN      PTR

;; ANSWER SECTION:
125.1.168.192.in-addr.arpa. 9   IN      PTR     WDMyCloud.fritz.box.

;; AUTHORITY SECTION:
125.1.168.192.in-addr.arpa. 9   IN      NS      fritz.box.

;; ADDITIONAL SECTION:
fritz.box.              9       IN      A       192.168.1.1

;; Query time: 2 msec
;; SERVER: 192.168.1.1#53(192.168.1.1)
;; WHEN: Mon Nov 23 19:56:49 GMT 2020
;; MSG SIZE  rcvd: 107
```
```
pi@raspberrypi:~ $ dig -x 192.168.1.32 @192.168.1.1

; <<>> DiG 9.11.5-P4-5.1+deb10u2-Raspbian <<>> -x 192.168.1.32 @192.168.1.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 27147
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0

;; QUESTION SECTION:
;32.1.168.192.in-addr.arpa.     IN      PTR

;; AUTHORITY SECTION:
1.168.192.in-addr.arpa. 9       IN      SOA     fritz.box. admin.fritz.box. 1606161417 21600 1800 43200 10

;; Query time: 1 msec
;; SERVER: 192.168.1.1#53(192.168.1.1)
;; WHEN: Mon Nov 23 19:56:57 GMT 2020
;; MSG SIZE  rcvd: 103
```
Screen from router:
![obraz](https://user-images.githubusercontent.com/716018/100009174-c82f7b80-2dce-11eb-8d79-1a4ea1e8fe53.png)

Screen from pi-hole:
![obraz](https://user-images.githubusercontent.com/716018/100009732-8b17b900-2dcf-11eb-8b39-9cf246944954.png)

**Comment by DL6ER** [comment]
The mesh configuration shouldn't be relevant here. However, we see that it is a problem of the Fritzbox, not your Pi-hole so I'm afraid we cannot do much/(anything?) to resolve this. (I'm only quoting the relevant lines below)

When you asked for the IP of `192.168.1.125`, both your Pi-hole and your Fritzbox knew the name:
```
pi@raspberrypi:~ $ dig -x 192.168.1.125 @127.0.0.1
125.1.168.192.in-addr.arpa. 9   IN      PTR     WDMyCloud.fritz.box.

pi@raspberrypi:~ $ dig -x 192.168.1.125 @192.168.1.1
125.1.168.192.in-addr.arpa. 9   IN      PTR     WDMyCloud.fritz.box.
```
---

However, when you asked for the device `192.168.1.32`, your Fritzbox did ***not*** reply with a name:
```
pi@raspberrypi:~ $ dig -x 192.168.1.32 @192.168.1.1
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN  <--- NXDOMAIN means "This device does not exist / has no name"
```
together with an empty answer:
```
;32.1.168.192.in-addr.arpa.     IN      PTR
```

It is clear that Pi-hole cannot know the name when your Fritzbox tells it that this device does not exist (and/or has no name).

Now we know that the issue is on the Fritzbox side because it knows the host name (according to your screenshot) but does not reply to name queries for this domain. You should contact AVM support about this because I really don't see anything we could do here. I'm sorry about that.

---

Concerning the address you observed for one device (`100.102.62.163`) I cannot comment much without knowing what device this is exactly (maybe not even then). The address comes from a reversed range with the description:
> 100.64.0.0 ~ 100.127.255.255 (100.64.0.0/10 prefix) carrier-grade NAT communication between service provider and subscribers

see https://www.speedguide.net/ip/100.102.62.163

---

I see that you're loosing capitalization between your Fritzbox and your Pi-hole. Even when domains are case-insensitive, I do see some value in keeping the capitalization to ease readability. I'll check if we can keep the capitalization and change all our comparisons to case-insensitive. It'll be a question of possible performance loss, but, at least for host names, we should be able to keep caps.

**Comment by Marooned-MB** [comment]
Thanks for all your support while going through it with me. Much appreciated!
I gained some knowledge on the way so it's a side bonus.

Thanks again.
Closing.

**Comment by DL6ER** [comment]
Capitalization of client names will be kept when https://github.com/pi-hole/FTL/pull/935 is merged.
